pkg/types/git_project.go

```diff
@@ -22,10 +29,30 @@ func (gp *GitProject) UnmarshalYAML(unmarshal func(interface{}) error) error {
 	return unmarshal((*raw)(gp))
 }
 
+// validateDirName evaluate directory name against forbidden characters
+func validateDirName(dirName string) bool {
+	return len(gitDirPatternNeg.Split(dirName, 2)) <= 1
+}
+
+// validateGitSubDir evaluate syntax for subdirectory path
+func validateGitSubDir(path string) bool {
+	var b = "/base"
+	for _, dirName := range strings.Split(path, "/") {
+		if ok := validateDirName(dirName); !ok {
+			return false
+		}
+		b = filepath.Join(b, dirName)
+		if b == "/" {
```

Comment:
No need to check for escaping here as this is done later when loading the project.